### ISSUE TITLE
aws_s3_bucket_object changing object encryption if no encryption set

### ISSUE THREAD

**Issue by zerfahp** [terraform-providers/terraform-provider-aws#10200]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

0.11.10

### Affected Resource(s)

aws_s3_bucket_object

### Terraform Configuration Files

```hcl
resource "aws_s3_bucket_object" "bootstrap_emr_object" {
    key = "${var.config_folder_path}/${var.cluster_name}/bootstrap-emr.sh"  
    bucket = "${var.config_bucket_id}"  
    content = ${data.template_file.emr_bootstrap_stack_template.rendered}"
}
```

### Expected Behavior

Object takes the default encryption key set on the bucket.

### Actual Behavior

Object takes the AWS/S3 KMS key for encryption and ignores the default encryption on the bucket

### Steps to Reproduce

1. 
I upgraded provider version from 0.2.19 to 0.2.20 and ran plan.

I get this in the plan...

 ~ module.spark_emr_cust_rels.aws_s3_bucket_object.bootstrap_emr_object
      kms_key_id:                                                          "arn:aws:kms:eu-west-1:367462524183:key/af5ae4a8-d4f6-4d77-91fa-4459XXXXXXX" => ""

2.
I confirm the plan and apply. KMS key on the object in AWS is now the key with alias "aws/s3" not the KMS SSE key the object used before and which is set as the default encryption on the bucket.

* #0000

**Comment by sbrandtb** [comment]
Happens on TF 0.12.9 and AWS provider 2.27.0 as well.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 3.10.0 of the Terraform AWS Provider, later this week. Thanks to @mattburgess for the implementation. 👍
